metaflow/plugins/kfp/tests/Dockerfile_kfp_step

```diff
@@ -0,0 +1,12 @@
+FROM analytics-docker.artifactory.zgtools.net/artificial-intelligence/ai-platform/aip-py39-cpu-cicd:4.0.1380
+ARG KFP_STEP_IMAGE_ARG
+ENV KFP_STEP_IMAGE=$KFP_STEP_IMAGE_ARG
+COPY . /home/zservice/metaflow
+RUN pip install --upgrade --force-reinstall -e /home/zservice/metaflow
+RUN pip install \
+    pytest==6.1.2 \
+    pytest-xdist==2.1.0 \
+    pytest-cov==2.10.1 \
+    subprocess-tee==0.3.2 \
+    pyyaml==5.4.1 \
+    kubernetes==18.20.0
```

Comment:
Curious why the same requirement `kubernetes==18.20.0` is not needed for the other Dockerfile. Would it be possible to use the same Dockerfile but passing in different `BASE_IMAGE_ARG`?

Reply:
If you can use the same docker file: Can you move the ARG lines (line 2 and 3) to the end of file so that more of the docker image layer (pip install lines) can share the same cache?
```
ARG KFP_STEP_IMAGE_ARG
ENV KFP_STEP_IMAGE=$KFP_STEP_IMAGE_ARG
```

Reply:
`kubernetes=18.20.0` is actually needed for `assert_step_image` test function. We actually import `kubernetes`.

I get your point on keeping the same Dockerfile, but actually we do want two different files. The reason is that if the Docker images contain the same data, they will have the same digest, and Kubelet won't even download the second because it will use the first that is already in cache :)

Reply:
Your image will still differ by an environment variable right?

Reply:
Moreso the addition of a new library (`kubernetes`).